MOBILEDOC.md

```diff
@@ -96,6 +96,29 @@ the start of the `value` text. As these tags are opened, then create a stack
 of opened markups. The `numberOfClosedMarkups` says how many markups should
 be closed at the end of a `value`.
 
+Text sections may contain [ATOMS](ATOMS.md) and markups. Atoms are self closing,
+and so the `numberOfClosedMarkups` will always be 0. Atoms have an additional
+`payload` entry in the array:
+
+```
+{
+  version: "0.3",
+  sections: [[
+    ["b"],                                ──── Bold Markup at index 0
+    ["i"],                                ──── Italic Markup at index 1
+    ["atom", "mention"]                   ──── Mention Atom at index 2
+  ],[
+    [openMarkupsIndexes, numberOfClosedMarkups, value, payload],
+    [[2], 0, "Bob", { id: 42 }],          ──── Mention Atom with payload
+    [[1], 0, "Example opening i tag"],
```

Comment:
From chat:

> @rlivsey hm, I imagined that they were a peer of markers
> and not markup at all
> it might require a type for markers

```
[0 /* means marker */, [2], 0, 'bob'],
[1 /* means atom */, 'mention', 'martin', { url: 'google.com }]
```

So split out the atom definitions into their own array too?

Eg something like:

```
{
  version: "0.3",
  sections: [[
    ["b"],         ──── Bold Markup at index 0
    ["i"],         ──── Italic Markup at index 1
  ],[
    ["mention"]    ──── Mention Atom at index 0
    ["thingy"]     ──── Thingy Atom at index 1
  ],[
    [0, [2], 0, 'bob'],                             ──── Italic Markup
    [1, 'mention', 'martin', { url: 'google.com' }] ──── Mention Atom
  ]]
}
```

As this is backwards incompatible anyway, would it be worth moving the definition to their own parts of the mobiledoc so the sections array is purely for the content?

Eg something like:

```
{
  version: "0.3",
  markups: [
    ["b"],         ──── Bold Markup at index 0
    ["i"],         ──── Italic Markup at index 1
  ],
  atoms: [
    ["mention"]    ──── Mention Atom at index 0
    ["thingy"]     ──── Thingy Atom at index 1
  ],
  sections: [
    [0, [2], 0, 'bob'],                             ──── Italic Markup
    [1, 'mention', 'martin', { url: 'google.com' }] ──── Mention Atom
  ]
}
```